packages/@aws-cdk/aws-codepipeline-actions/lib/servicecatalog/deploy-action.ts

```diff
@@ -0,0 +1,99 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import { Role, PolicyStatement, ManagedPolicy } from '@aws-cdk/aws-iam';
+import { Construct } from '@aws-cdk/core';
+import { Action } from '../action';
+
+/** TODO:
+ *  1.) Support cross account deployments
+ *  2.) Fix least privilege
+ *  3.) Handle CREATION of a new product
+ *  4.) Handle MAINTENANCE of a provisioned product
+ *  5.) Test/support product types beyond CLOUD_FORMATION_TEMPLATE
+ *  6.) Valid test cases!
+ */
+
+/**
+ * Construction properties of the {@link ServiceCatalogDeployAction ServiceCatalog server deploy CodePipeline Action}.
+ */
+export interface ServiceCatalogDeployActionProps extends codepipeline.CommonAwsActionProps {
+  /**
+   * The name of the template file found in the artifact to install as a new version of the Service Catalog Product
+   */
+  readonly templateFile: string;
+  /**
+   * The name of the VERSION of the Service Catalog product
+   */
+  readonly scProductVersionName: string;
+  /**
+   * The name of the PRODUCT TYPE of the Service Catalog product (this should probably be an ENUM, or implicit)
+   * Valid types CLOUD_FORMATION_TEMPLATE
+   * Possibly we can support MARKETPLACE in a future release.
+   */
+  readonly scProductType: string;
+  /**
+   * The DESCRIPTION of the VERSION of the Service Catalog product
+   */
+  readonly scProductVersionDescription: string;
```

Comment:
No `sc` prefix, and this property is optional according to the CodePipeline docs:

```suggestion
  readonly productVersionDescription:? string;
```